sklearn/utils/_array_api.py

```diff
@@ -1101,3 +1101,13 @@ def _tolist(array, xp=None):
         return array.tolist()
     array_np = _convert_to_numpy(array, xp=xp)
     return [element.item() for element in array_np]
+
+
+def _allclose(x, y, rtol=1e-5, atol=1e-8, xp=None):
+    xp, _ = get_namespace(x, y, xp=xp)
+    if hasattr(xp, "allclose"):
+        return xp.allclose(x, y, rtol=rtol, atol=atol)
```

Comment:
I have mixed feelings about relying on symbols that are not part of the spec, in particular since the numpy API and semantics are not consensual:

https://data-apis.org/array-api/latest/search.html?q=allclose
https://github.com/data-apis/array-api/issues/170

I think we should always convert to numpy here.